server/src/main/java/org/opensearch/search/aggregations/metrics/PercentileRanksAggregationBuilder.java

```diff
@@ -83,6 +83,7 @@ private PercentileRanksAggregationBuilder(String name, double[] values, Percenti
 
     public PercentileRanksAggregationBuilder(StreamInput in) throws IOException {
         super(in);
+        this.valuesField = VALUES_FIELD;
```

Comment:
It's a bit weird that these child classes have to know to assign this field or else things don't work correctly. I believe the two child classes you modified here are the only places this constructor is called? What do you think about modifying the parent constructor to take a `StreamInput in` along with a `ParseField valuesField` so that the class contract makes it clear that field needs to be defined by the implementer? Then you just call `super(in, VALUES_FIELD)` here.

Reply:
sure @andrross. this is the alternative solution i thought about. i'll make this change & push it now.